CDP4Orm/Dao/Supplemental/IterationSetupDao.cs

```diff
@@ -106,5 +125,32 @@ public IEnumerable<IterationSetup> ReadByEngineeringModelSetup(NpgsqlTransaction
                 }
             }
         }
+
+        /// <summary>
+        /// Execute additional logic before each delete function call.
+        /// </summary>
+        /// <param name="transaction">
+        /// The current transaction to the database.
+        /// </param>
+        /// <param name="partition">
+        /// The database partition (schema) where the requested resource will be deleted.
+        /// </param>
+        /// <param name="iid">
+        /// The thing DTO id that is to be deleted.
+        /// </param>
+        /// <param name="isHandled">
+        /// Logic flag that can be set to true to skip the generated deleted logic
+        /// </param>
+        /// <returns>
+        /// True if the concept was deleted.
+        /// </returns>
+        /// <remarks>
+        /// IterationSetups cannot be deleted at all.
+        /// </remarks>>
+        public override bool BeforeDelete(NpgsqlTransaction transaction, string partition, Guid iid, out bool isHandled)
+        {
+            isHandled = true;
```

Comment:
why do we have this override?

Reply:
I think this is the only way to jump out of a delete prematurely.

Reply:
I think @alexatrhea is right.